src/SonarScanner.MSBuild.PreProcessor/ProcessedArgs.cs

```diff
@@ -124,6 +129,14 @@ public string PropertiesFileName
             var isHostSet = AggregateProperties.TryGetValue(SonarProperties.HostUrl, out var sonarHostUrl); // Used for SQ and may also be set to https://SonarCloud.io
             var isSonarcloudSet = AggregateProperties.TryGetValue(SonarProperties.SonarcloudUrl, out var sonarcloudUrl);
             this.sonarServer = GetSonarServer(logger, isHostSet, sonarHostUrl, isSonarcloudSet, sonarcloudUrl);
+            ApiBaseUrl = AggregateProperties.TryGetProperty(SonarProperties.ApiBaseUrl, out var apiBaseUrl)
+                ? apiBaseUrl.Value
+                : SonarServer switch
+                {
+                    SonarCloudServer => "https://api.sonarcloud.io", // SC default
```

Comment:
I would move this to the Server class. It can be a method or a property and it will remove the need to check for the server type.